### PR TITLE
bindings: Add support for `Delete` for deleting manifest list from local storage.

### PR DESCRIPTION
**Pull request by flouthoc**
Bindings already support `Remove` which removes a manifest from the list
following function adds support for removing entire manifest for local
storage.

Similar functionality can be also used indirectly by using `Remove` defined in
image bindings

Closes: https://github.com/containers/podman/issues/14735

#### Does this PR introduce a user-facing change?

```
Bindings: Manifest package now exposes ManifestRemove to remove manifest list from local storage
```

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
I don't think this is really needed. podman-remote can already do this so no new code should be needed. To remove a manifest you just remove the image: https://github.com/containers/podman/blob/3176b3fe011aa3323ba192e90efe516d0701e46e/pkg/domain/infra/tunnel/manifest.go#L94-L97

**Comment by flouthoc** [comment]
@Luap99 Not sure but I think there is a issue with the way `podman-remote` is doing `manifest rm` it does not sets `LookupManifest` while making actual remove call see https://github.com/containers/podman/blob/main/pkg/domain/infra/tunnel/images.go#L31 also its not giving a good UX for `pkg/bindings/manifest` so adding this new API should make it more clean and once merged `manifest-rm` for remote can start using this instead of wiring `LookupManifest` across the API.

But i already mentioned that this just an easier alternative here https://github.com/containers/podman/issues/14735#issuecomment-1167170012 and but i'm also fine to keep older API as is.

**Comment by rhatdan** [comment]
@vrothberg @baude @jwhonce WDYT?

**Comment by vrothberg** [comment]
It screams for a better regression test as well.

**Comment by flouthoc** [comment]
> It screams for a better regression test as well.

@vrothberg I agree I think `podman manifest rm` should test if it resolves to manifest list instead of image when both exists before removing. Could you let me know if we need any new test in this PR or improve test for `podman-remote manifest rm` when we instrument that I think you meant the latter one but just confirming. 

I see two followup PR
* Fix `images` API and `tunnel) Remove(` so it accepts and honor `LookupManifest`
* Instrument `podman-remote manifest rm` to start using new API in bindings ( preferably after we fix existing API )

**Comment by vrothberg** [comment]
> > It screams for a better regression test as well.
> 
> @vrothberg I agree I think `podman manifest rm` should test if it resolves to manifest list instead of image when both exists before removing. Could you let me know if we need any new test in this PR or improve test for `podman-remote manifest rm` when we instrument that I think you meant the latter one but just confirming.

Whatever tests we add should fail with a previous version of Podman.  Since the LookupManifest field is ignored, we need a test that would fail if it's unset. Presumably when there's a local manifest with a matching image.  In that case, the matching image should be removed prior to the fixes (and not the manifest).

> I see two followup PR
> 
>     * Fix `images` API and `tunnel) Remove(` so it accepts and honor `LookupManifest`

Locally it should work. Remotely it needs fixing.

>     * Instrument `podman-remote manifest rm` to start using new API in bindings ( preferably after we fix existing API )

Yes, that sounds good to me.

**Comment by flouthoc** [comment]
@vrothberg Yes SGTM I'll address the test fixup in a follow PR for `podman-remote manifest rm` which will actually exercise the test. Since this is bindings PR I have already added a test for this. Let me know if anything else needs to be done on this PR.

**Review by jwhonce — CHANGES_REQUESTED** [review]
Exposing existing API is fine, one change please.

**Comment by jwhonce** [comment]
/hold

pending method rename.

File: pkg/bindings/manifests/manifests.go
Comment (flouthoc) [reply pkg/bindings/manifests/manifests.go]: Done :)

**Comment by rhatdan** [comment]
LGTM
@nalind @vrothberg @giuseppe @Luap99 PTAL

File: pkg/bindings/manifests/manifests.go
Comment (Luap99) [inline pkg/bindings/manifests/manifests.go:121]: Why does it return an array of errors? Will there ever be a case where it returns more than one error?

**Review by Luap99 — COMMENTED** [review]
Please update your commit/PR title with the new function name.

File: pkg/bindings/manifests/manifests.go
Comment (flouthoc) [reply pkg/bindings/manifests/manifests.go:121]: @Luap99 Since API returns a list of errors so just making sure bindings honor it we can match the struct to upstream API if we decide to change the API https://docs.podman.io/en/latest/_static/api.html?version=v4.0#operation/ManifestDeleteLibpod

Comment (rhatdan) [reply pkg/bindings/manifests/manifests.go:121]: You need to wait til 5.0 to break existing API.

Comment (giuseppe) [reply pkg/bindings/manifests/manifests.go:121]: can we use `pkg/errorhandling/JoinErrors` to get a single error?

Comment (flouthoc) [reply pkg/bindings/manifests/manifests.go:121]: @giuseppe SGTM we can do this at binding end.

Comment (flouthoc) [reply pkg/bindings/manifests/manifests.go]: > can we use `pkg/errorhandling/JoinErrors` to get a single error?

Amended with this suggested change.

**Pull request merged**
